### PR TITLE
odim-105 unit test of plugin with duplicate manager addr

### PR DESCRIPTION
**Pull request by rameshkk-hpe**

### CODE REVIEW

File: svc-aggregation/rpc/common.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func GetAggregator() *Aggregator {
 			EventNotification:       agmessagebus.Publish,
 			GetAllKeysFromTable:     agmodel.GetAllKeysFromTable,
 			GetConnectionMethod:     agmodel.GetConnectionMethod,
-			GetPluginData:		 agmodel.GetPluginData,
+			GetPluginMgrAddr:	 agmodel.GetPluginMgrAddr,
```
Comment (jeevan-kamkar) [inline svc-aggregation/rpc/common.go:58]: this actual function agmodel why this name is changed

File: svc-aggregation/agmodel/system.go [hunk 2]
```diff
@@ -160,6 +160,21 @@ func GetPluginData(pluginID string) (Plugin, *errors.Error) {
 	return plugin, nil
 }
 
+func GetPluginMgrAddr(pluginID string) (Plugin, *errors.Error) {
```
Comment (jeevan-kamkar) [inline svc-aggregation/agmodel/system.go:163]: why this function added in agmodel code with random port, id, and host

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
Address  the changes

File: svc-aggregation/rpc/common.go (hunk 1 above)
Comment (rameshkk-hpe) [reply svc-aggregation/rpc/common.go:58]: I earlier committed changes to use mockup function for GetPluginData in UT. Now replaced that call with call to new function GetPluginMgrAddr()

Comment (shivacharanms) [reply svc-aggregation/rpc/common.go:58]: I think the question was why this change

File: svc-aggregation/agmodel/system.go (hunk 2 above)
Comment (shivacharanms) [reply svc-aggregation/agmodel/system.go:163]: yes i agree, we cannot use dummy values in agmodel.

File: svc-aggregation/rpc/common.go (hunk 1 above)
Comment (rameshkk-hpe) [reply svc-aggregation/rpc/common.go:58]: GetPluginData is an existing function widely used and mocking it is interfering with existing UTs. So created GetPluginMgrAddr which is mocked in UT (specifically for this fix) and just wraps GetPluginData in normal execution.

File: svc-aggregation/agmodel/system.go (hunk 2 above)
Comment (rameshkk-hpe) [reply svc-aggregation/agmodel/system.go:163]: Changed; hardcoding removed & tested.

**Review by jeevan-kamkar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**